lib/_http_outgoing.js

```diff
@@ -553,6 +554,19 @@ OutgoingMessage.prototype.write = function write(chunk, encoding, callback) {
 };
 
 function write_(msg, chunk, encoding, callback, fromEnd) {
+  if (msg.closed) {
```

Comment:
I don't think this will actually solve the problem. The issue happens when the server destroys the socket and client tries to write to the destroyed socket.

For the client the socket could be still alive (no `'close'` emitted) so this property might be not set when the bad write occurs.